packages/storage/src/implementation/list.ts

```diff
@@ -24,26 +24,41 @@ import * as json from './json';
 import * as type from './type';
 import { ListResult } from '../list';
 
-export function fromResource(
+interface ListMetadataReource {
+  name: string;
+  bucket: string;
+}
+
+interface ListResultResource {
+  prefixes: string[];
+  items: ListMetadataReource[];
+  nextPageToken?: string;
+}
+
+const maxResultsKey = 'maxResults';
```

Comment:
I believe you should use all uppercase here: https://g3doc.corp.google.com/javascript/typescript/g3doc/styleguide.md?cl=head#identifiers